### PR TITLE
Update rust toolchain version

### PR DESCRIPTION
**Pull request by maxammann**
Update to 0.62.0 and latest nightly

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [toolchain]
-channel = "1.60"
+channel = "1.62"
```
Comment (nyurik) [inline rust-toolchain.toml:2]: Can this be set to "stable" instead?

**Review by nyurik — APPROVED** [review]

File: rust-toolchain.toml (hunk 1 above)
Comment (maxammann) [reply rust-toolchain.toml:2]: I think its nice to opt-into new versions. That way we we have always defined which minimal rust version is required to build the project.

In the future we maybe stick with older versions of rust (potentially).

Comment (nyurik) [reply rust-toolchain.toml:2]: I would have kept it on stable until something breaks, and then switch to a fixed version if needed, but up to you

Comment (maxammann) [reply rust-toolchain.toml:2]: We are also on a defined nightly version. Both should be updated at the same time. I think this way we have more control. But maybe we find some recommendation for this.

Comment (maxammann) [reply rust-toolchain.toml:2]: Im not sure how cargo handles the following case:

A user has an old stable installed, then checks out the project, which requires a newer Rust version. Does it update the stable when `cargo build` is executed?

Comment (nyurik) [reply rust-toolchain.toml:2]: Sure, your call :) I don't think cargo will download the latest on build.

**Pull request merged**